src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/MergeNestedIfStatementsTests.cs

```diff
@@ -0,0 +1,2617 @@
+            System.Console.WriteLine(b);
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedIntoElseIfWithUnmatchingElseClauses2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+            System.Console.WriteLine();
+        else if (a)
+        {
+            [||]if (b)
+                System.Console.WriteLine(a && b);
+            else
+                System.Console.WriteLine(a);
+        }
+        else
+            System.Console.WriteLine(b);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedIntoElseIfWithUnmatchingElseClauses3()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+            System.Console.WriteLine();
+        else if (a)
+        {
+            [||]if (b)
+                System.Console.WriteLine(a && b);
+            else
+            {
+                System.Console.WriteLine(a);
+            }
+        }
+        else
+            System.Console.WriteLine(b);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedIntoElseIfWithUnmatchingElseClauses4()
+        {
+            // Do not consider the using statement to be a simple block (as might be suggested by some language-agnostic helpers).
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+            System.Console.WriteLine();
+        else if (a)
+        {
+            [||]if (b)
+                System.Console.WriteLine(a && b);
+            else
+            {
+                System.Console.WriteLine(a);
+            }
+        }
+        else
+            using (null)
+                System.Console.WriteLine(a);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedWithMatchingElseClauses1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            [||]if (b)
+                System.Console.WriteLine(a && b);
+            else
+                System.Console.WriteLine(a);
+        }
+        else
+            System.Console.WriteLine(a);
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a && b)
+            System.Console.WriteLine(a && b);
+        else
+            System.Console.WriteLine(a);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedWithMatchingElseClauses2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            [||]if (b)
+                System.Console.WriteLine(a && b);
+            else
+            {
+                System.Console.WriteLine(a);
+            }
+        }
+        else
+        {
+            System.Console.WriteLine(a);
+        }
```

Comment:
same applies here.  i feel like a small set of tests that validate that braces do/dont' matter for each would be fine.  Thhat validates the brace-handling (or 'statement-finding' portion of the code).  Then you can have functionality tests like "what are the cases where we shoud not match, and the cases where we should

Right now, these seem to just be about taking the few decisions that the code has to make and permuting each one of them.  I'd rather have each decision well tested, with then a tiny amount of additional cases to handle the variants for the other decisions.